gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -155,20 +159,65 @@ private List<Reference> getSpecimenReferences(RCMRMT030101UKCompoundStatement co
     }
 
     private void setResultReferences(RCMRMT030101UKCompoundStatement compoundStatement, DiagnosticReport diagnosticReport) {
+        var resultReferences = getDirectResultReferences(compoundStatement);
 
-        var resultReferences = compoundStatement.getComponent()
-            .stream()
-            .filter(RCMRMT030101UKComponent02::hasNarrativeStatement)
-            .map(RCMRMT030101UKComponent02::getNarrativeStatement)
-            .filter(narrativeStatement -> !narrativeStatement.getText().contains(LAB_REPORT_COMMENT_TYPE))
-            .map(narrativeStatement -> new Reference(new IdType(ResourceType.Observation.name(), narrativeStatement.getId().getRoot())))
-            .collect(toCollection(ArrayList::new));
+        // also include USER COMMENT or UNKNOWN references from nested BATTERY as per NIAD-2666
```

Comment:
Oh gosh, I'm not a huge fan of referencing Jira tickets for useful context, as it's my belief that code lives longer than Jira. Also, this code is open source whereas Jira is closed off.

Is there anything useful in Jira which should be pulled into this comment? Ideally the tests should give the most information.

Reply:
removed this as is no longer required